src/core/server/ui_settings/ui_settings_client.ts

```diff
@@ -143,8 +146,17 @@ export class UiSettingsClient implements IUiSettingsClient {
   }
 
   private async getRaw(): Promise<UiSettingsRaw> {
+    const cachedValue = this.rawCache.get();
+    if (cachedValue) {
+      return cachedValue;
+    }
+
     const userProvided = await this.getUserProvided();
-    return defaultsDeep({}, userProvided, this.defaults);
+    const result = defaultsDeep({}, userProvided, this.defaults);
```

Comment:
This pattern of using lodash with 3 arguments is exactly what we are trying to avoid. It really must be two arguments.

Reply:
What pattern do you suggest using when merging two objects into a third one without mutating any of the input objects then?

Reply:
IMO the best alternative would be don't use `defaultsDeep` at all. If we removed `getAll` from the public contract, we would reduce the number of js object processed on every `get(..)`.